### ISSUE TITLE
[Bug] [Survival] Cannot move items between player inventory, assembler, and crucible

### ISSUE THREAD

**Issue by asanagisae** [tsunamayo/Starship-EVO#5887]
### Short Description of the issue:

I started a new survival world in 24w29b. A few issues make the experience frustrating.

I am unable to retrieve ores, unspent coal, or ingots from the crucible block. When you click on one, you can hold it, but you cannot put it into your hotbar. You can only throw them on the ground, then retrieve them after.

I am unable to put ingots into the input for the assembler. I cannot build a cargo block. I cannot feed the assembler via tubes from cargo blocks (because I can't build cargo blocks). The assembler therefore cannot build anything.

### Steps to reproduce the issue:

Setup:

1. Create a new survival world
2. Gather a small quantity of each ore type
3. Produce a work bench, a crucible, an assembler, or other similar blocks
4. Place a starter block and put the above building blocks on a simple starter base
5. Attempt to smelt or create basic things with your assemblers

It is possible to put ores and coal into the crucible, and the crucible will create ores. However, _retrieving_ them from the crucible is difficult.

Things I cannot do:

1. Take ores/ingots from crucible or assembler, and put them in player inventory - I can only throw them out first
2. Put ingots into assembler's input slots - I can only put them in storage
3. Assemblers will not take ingots from the storage area to build parts or blocks
4. Assemblers will not queue up necessary components to build blocks (like plates for cargo)

Having a recipe queued (like cargo blocks) still does not enable the ability to put ingots into the assembler's input slots.

Also, if you right-click a crucible that is currently smelting (it has ores, ingots, and coal) none of the items inside will be dropped - you will lose your ores, ingots, and coal that were in the crucible.

### Environment
- Starship EVO version: 24w29b

### Additional information:

https://github.com/user-attachments/assets/9d5d6a06-3831-4ba8-8b31-ae12f05b5e12

https://github.com/user-attachments/assets/f9d811f0-e9da-4e05-9123-4befffe979d7

I have no reason to suspect that this issue is specific to my world, but I have attached the world for convenience.

[Sky Wanderer.zip](https://github.com/user-attachments/files/16272957/Sky.Wanderer.zip)

You should spawn right beside a small base. To reproduce the issues, try to retrieve the ores and ingots from the crucible. Try to put ingots into the assembler to create a cargo container.

These basic steps appear to be impossible in survival currently.

These issues persist after reloading the world.

**Comment by asanagisae** [comment]
I took a second look at this situation.

So the assembler not working is due to the base not having power. I added a windmill and that was solved.

I suspect the inability to put ingots in my inventory is related to the backpack thinking it's "beyond full", at 98,119 L out of 3,000 L. I moved all the items from my backpack to a cargo container and it still reports this odd number.

![image](https://github.com/user-attachments/assets/5d831a1f-45d6-44ad-bd17-58fc65f6c18b)

When I threw this light block out of my inventory, the backpack dropped to 17,872 L. I picked it up from right-clicking on a nearby wreck. I wonder if those blocks are "corrupted" somehow?

![image](https://github.com/user-attachments/assets/34b354f7-84a9-426b-91dd-882a3b7cd677)

**Comment by asanagisae** [comment]
There's possibly one more issue. When I mine, often I will end up over 100% full on ores and stone. I move a stack of stone to an arc furnace, and it starts to smelt. I then move the stack of iron ores into the arc furnace instead, and the stack of stone moves to my hand. I can't put it in my inventory because it would result in being over 100% full again. If I open the "transfer items" menu on the arc furnace, I end up throwing all the stone on the ground.

![image](https://github.com/user-attachments/assets/714b4932-4656-46e1-8e61-88957d0c5953)

This stone can't be picked up. It's several hundred stone and I need silicon....

If I drop ores or ingots this way, those can be picked up.

**Comment by tsunamayo** [comment]
@asanagisae Thanks, also I should not you are not supposed to put stuff into the input slots of an assembler. You simply select the blueprint you want to craft.

**Comment by tsunamayo** [comment]
@asanagisae Also I would like to know how you manage to pick a block? It should be impossible on wreck. Can you send me your world and tell me where is that wreck? Thanks

**Comment by tsunamayo** [comment]
To be more precise, I dont understand how it working: " (hold a crafted block and right-click on the wreck as if you built it) ", if you could send a video or detail more so I can fix the issue.
You are not supposed to be able to build a block on a wreck, I could not in my test.

**Comment by asanagisae** [comment]
I'm not _adding_ to a wreck. I mean like this:

https://github.com/user-attachments/assets/67849289-188e-4ae4-9710-e84b689c1af1

Although now that you mention it....

https://github.com/user-attachments/assets/97fd0b40-81c0-456e-9155-2e6ed914f118

This is a brand new survival world. Started just now. Did everything above in the first and only play session, in the first 5 minutes.

**Comment by tsunamayo** [comment]
@asanagisae I still cant reproduce. Do you get this in your first session only, or do you get this also after a reload? Thanks

**Comment by asanagisae** [comment]
@tsunamayo I get it in the first session after creating the world and after a reload. Every time I start the game up, I can go up to a wreck and steal 95% of the blocks on it. I've never had a wreck where I couldn't steal blocks from it if I tried.

I've noticed that sometimes, when I right-click a block, the block will vanish and not produce any items (like creative mode). Sometimes the block _will_ produce an item for it and I pick it up (like survival). This happens both for ships/stations I build and for wrecks I find.

This does mean that I sometimes lose legitimate blocks that I farmed ores for and crafted on my ships. And sometimes I right-click a block on a wreck and the block disappears without dropping an item.

I'm not sure what the trigger is for these inconsistencies.
